.ci/s390x/lib_install_qemu_s390x.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 IBM
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -e
+
+CURRENT_QEMU_VERSION=$(get_version "assets.hypervisor.qemu.version")
+PACKAGED_QEMU="qemu"
+
+get_packaged_qemu_version() {
+        if [ "$ID" == "ubuntu" ]; then
```

Comment:
Idea: I'd be tempted to deal with the error scenario first, so in each function:

```sh
[ "$ID" == "ubuntu" ] || die "Unsupported distro: $ID"

 :
```

That will also simplify the remainder of the code as you can remove one layer of `if` testing. Might be worth putting that `die` test at the top global level to avoid having to duplicate either the code in each function, or duplication a call to a function that runs that test?

Reply:
Wouldn't this just mean that i check the content of $ID at two places? I would instead use a switch case to replace this if elif structures and have the error scenario as a default.

Reply:
Well, as I say, you could make this a global check in `.ci/s390x/lib_install_qemu_s390x.sh`. Something like:

```sh
[ "$ID" == "ubuntu" ] || die "Unsupported distro: $ID"

get_packaged_qemu_version() {
   # ...
}
```

I think that should work since the caller already sources `/etc/os-release` and `lib.sh` which gives access to `die()`.

Reply:
Yes, it should work at that position. I changed it in the code.